client/src/main/java/io/pravega/client/ClientFactory.java

```diff
@@ -73,6 +74,7 @@ static ClientFactory withScope(String scope, URI controllerUri) {
      * @return Instance of ClientFactory implementation.
      */
     static ClientFactory withScope(String scope, ClientConfig config) {
+        config = new CredentialsExtractor().extract(config);
```

Comment:
This same code is duplicated in a few places. Can we add a static method to do this?

Reply:
Replaced with static method call.

Reply:
Using `toBuilder` lombok functionality as a copy constructor.